lib/internal/child_process.js

```diff
@@ -352,10 +352,10 @@ function onErrorNT(self, err) {
 ChildProcess.prototype.kill = function(sig) {
   var signal;
 
-  if (sig === 0) {
-    signal = 0;
-  } else if (!sig) {
+  if (!sig) {
```

Comment:
Ouch! That's totally wrong. I didn't test `0` after making the suggested change. Will fix.